### PR TITLE
Added optional parameter initPayload

### PR DESCRIPTION
**Pull request by lordgreg**
This PR adds the optional parameter to the static `connect` function. initPayload is a type of `Map<String, String>`, which will then be converted to JSON format. As specified by Apollo Client and required subscription authorisation, we could then simply do:

```
  Map<String, String> initPayload = new Map();
  initPayload['accessToken'] = Auth.accessToken;

    await SocketClient.connect(
      Static.BASE_URL_GRAPHQL_SUB,
      initPayload: initPayload
    )
    .then((createdClient) {
      client = createdClient;
      connections[operationName] = client;
    });
```

That being used, `connection_init` should now send the payload with `accessToken` to the websocket stream.

Resolves: https://github.com/zino-app/graphql-flutter/issues/74

### CODE REVIEW

File: lib/src/websocket/messages.dart [hunk 1]
```diff
@@ -45,12 +47,22 @@ abstract class GraphQLSocketMessage extends JsonSerializable {
...
+  final Map<String, String> payload;
+
+  InitOperation(this.payload) : super(MessageTypes.GQL_CONNECTION_INIT);
 
   @override
-  dynamic toJson() => {
-        'type': type,
-      };
+  dynamic toJson() {
+    final Map<String, String> jsonMap = new Map();
```
Comment (HofmannZ) [inline lib/src/websocket/messages.dart]: No need for the `new` keyword.

File: lib/src/websocket/messages.dart [hunk 2]
```diff
@@ -45,12 +47,22 @@ abstract class GraphQLSocketMessage extends JsonSerializable {
...
+    jsonMap['type'] = type;
+
+    if (this.payload != null) {
+      jsonMap['payload'] = json.encode(this.payload);
+    }
+
+    return  json.encode(jsonMap);
+
```
Comment (HofmannZ) [inline lib/src/websocket/messages.dart]: No need for the new line here.

File: lib/src/socket_client.dart [hunk 3]
```diff
@@ -16,22 +16,26 @@ class SocketClient {
...
     },
+    final Map<String, String> initPayload
   }) async {
+    _initPayload = initPayload;
+
     return SocketClient(GraphQLSocket(await WebSocket.connect(
       endPoint,
       protocols: protocols,
-      headers: headers,
+      headers: headers
```
Comment (HofmannZ) [inline lib/src/socket_client.dart]: No need to remove the comma here.

**Comment by lordgreg** [comment]
Hi @HofmannZ,

I've resolved the issues you were commenting. Also, I've moved my PR into branch next instead of master.

**Comment by HofmannZ** [comment]
@lordgreg Changes look good! Only two merge conflicts that need to be resolved before I can merge. Also try to run flutter lint in the project before you commit, that will do all the formatting for you.

**Comment by lordgreg** [comment]
Reminder to myself:

Next time, instead of trying to merge in master, just create a new branch from a branch you have to push into. What a mess :)

Sorry for it.

**Comment by HofmannZ** [comment]
@lordgreg No problem! Looks good now, well be included in the next release

**Pull request merged**